src/MongoDB.Driver.Core/Core/NativeLibraryLoader/LibraryLoader.cs

```diff
@@ -33,7 +33,10 @@ internal class LibraryLoader
         public LibraryLoader(ILibraryLocator libraryLocator)
         {
             Ensure.IsNotNull(libraryLocator, nameof(libraryLocator));
-            ThrowIfNot64BitProcess();
+            if (!libraryLocator.IsX32ModeSupported)
```

Comment:
This is a temporal property to support x32 mode with snappy.

Reply:
cc: @rstam

Reply:
no problem